storageClient/storageClient.go

```diff
@@ -83,6 +83,18 @@ func (c *StorageClient) DeleteGroup(ctx context.Context, params DeleteGroupReque
 	return err
 }
 
+// ListGroups returns all the groups a client is associated with (if any).
```

Comment:
`returns all groups for the client's account by default or optionally filter by parameters such as client ID`